src/views/TimeOutWrapper/index.tsx

```diff
@@ -12,27 +17,163 @@ const sessionTimeout = { minutes: 14 };
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const sessionInterval = useRef<any>();
+  const [timeRemainingArr, setTimeRemainingArr] = useState([0, 'second']);
   const lastActiveAt = useSelector(
     (state: AppState) => state.auth.lastActiveAt
   );
 
-  const { authService }: { authService: any } = useOktaAuth();
-  registerExpire(authService, lastActiveAt);
+  const sessionExpiration = useSelector(
+    (state: AppState) => state.auth.sessionExpiration
+  );
+
+  const oneSecond = Duration.fromObject({ seconds: 1 }).as('milliseconds');
+  const oneMinute = Duration.fromObject({ minutes: 1 }).as('milliseconds');
+  const fiveMinutes = Duration.fromObject({ minutes: 5 }).as('milliseconds');
+
+  const storeSession = async () => {
+    // eslint-disable-next-line no-underscore-dangle
+    const session = await authService._oktaAuth.session.get();
+    const expiresAt = new Date(session.expiresAt).getTime();
+    dispatch(updateLastSessionRenew(expiresAt));
+  };
+
+  const formatSessionTimeRemaining = (): [number, string] => {
+    if (sessionExpiration) {
+      const currentTime = Date.now();
+      const timeRemaining = sessionExpiration - currentTime;
+
+      // If over a minute remaining, displays X minutes
+      if (timeRemaining > oneMinute + oneSecond) {
+        // We need to add a minute to make sure the correct number of minutes displays
+        // e.g. 4 minutes 59 seconds shows 4 minutes; add a minute to display 5 minutes
+        const minutes = Duration.fromMillis(timeRemaining + oneMinute).as(
+          'minutes'
+        );
+        return [Math.floor(minutes), 'minute'];
+      }
+
+      if (timeRemaining >= oneMinute) {
+        return [1, 'minute'];
+      }
+
+      // If less than a minute remaining, display X second(s)
+      const seconds = Duration.fromMillis(timeRemaining).as('seconds');
+      return [Math.floor(seconds), 'second'];
+    }
+    return [0, 'second'];
+  };
+
+  const handleModalExit = () => {
+    clearTimeout(sessionInterval.current);
+    setIsModalOpen(false);
+    storeSession();
+  };
+
+  useInterval(
+    () => {
+      const currentTime = Date.now();
+
+      if (currentTime <= sessionExpiration) {
+        setTimeRemainingArr(formatSessionTimeRemaining());
+      }
+    },
+    isModalOpen ? 1000 : null
+  );
+
+  // If the session expires in 5 minutes or less
+  useInterval(
+    () => {
+      const currentTime = Date.now();
+      if (currentTime + fiveMinutes > sessionExpiration) {
+        setTimeRemainingArr(formatSessionTimeRemaining());
+        setIsModalOpen(true);
+      }
+    },
+    sessionExpiration && !isModalOpen ? 1000 : null
+  );
+
+  useInterval(
+    () => {
+      const currentTime = Date.now();
 
-  return <>{children}</>;
+      // If the session expired
+      if (currentTime >= sessionExpiration) {
+        clearInterval(sessionInterval.current);
+        setIsModalOpen(false);
+        authService.logout('/login');
```

Comment:
Why are we manually logging the user out? If the session expires, then we shouldn't need to log out, right?